library/src/main/java/com/billing/google/GoogleBillingService2.kt

```diff
@@ -115,8 +115,11 @@ class GoogleBillingService2(val context: Context, private val inAppSkuKeys: List
             }
             BillingClient.BillingResponseCode.USER_CANCELED ->
                 log("onPurchasesUpdated: User canceled the purchase")
-            BillingClient.BillingResponseCode.ITEM_ALREADY_OWNED ->
+            BillingClient.BillingResponseCode.ITEM_ALREADY_OWNED -> {
                 log("onPurchasesUpdated: The user already owns this item")
+                //item already owned? call queryPurchases to verify and process all such items
+                queryPurchases()
+            }
```

Comment:
This solution is copied from codelabs.developers.google.com
https://codelabs.developers.google.com/codelabs/play-billing-scalable-kotlin/index.html?index=..%2F..index#3

Idea: let's double-check Google's cache if we receive that type of message and unlock all needed items.

Reply:
We tried something similar before. It seemed that the problem was that the local cache was NOT updated, but when the user tried to purchase, it verified on the server and failed because it was already purchased. Customers found that it "fixed itself" in in about 24 hours, but it was frustrating all around until that time passed.

Let's try this and see. I suspect (hope!) that because we've adopted the new API, that Google is much more consistent with this and the cache will now reflect what the server sees more consistently.